### PR TITLE
HPCC-17866 Document upper size limit for DATA/STRING types

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

@ghalliday Please review 

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: docs/ECLLanguageReference/ECLR_mods/Basics-Constants.xml [hunk 1]
```diff
@@ -20,6 +20,10 @@
     string, prepend a backslash (\). To include the backslash character (\) in
     a constant string, use two backslashes (\\) together.</para>
 
+    <para>The upper size limit for any string constant is 4GB (2^32), although
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/Basics-Constants.xml]: I don't think the extra comment about running out of memory adds anything.  I'm not sure it is worth documenting the maximum length of a string constant at all, no one is going to have a string constant that long.

File: docs/ECLLanguageReference/ECLR_mods/Value-Data.xml [hunk 2]
```diff
@@ -24,6 +24,9 @@
   role="bold">Working with BLOBs</emphasis> for more information on this
   subject.</para>
 
+  <para>The upper size limit for any DATA value is 4GB (2^32), although you
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/Value-Data.xml]: I would remove the comment about running out of memory.  Same goes for all the other items.

File: docs/ECLLanguageReference/ECLR_mods/Value-QString.xml [hunk 3]
```diff
@@ -23,6 +23,9 @@
   to the size needed to contain the result of a cast or passed parameter. You
   may use set indexing into any QSTRING to parse out a substring.</para>
 
+  <para>The upper size limit for any QSTRING value is 4GB (2^32), although you
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/Value-QString.xml]: Limit is probably 2^32 characters (which is smaller).

File: docs/ECLLanguageReference/ECLR_mods/Value-Data.xml (hunk 2 above)
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/Value-Data.xml]: Actually the limits are probably 2^32-17!
I wouldn't document that, but it also may not be worth saying 2^32.

**Review by ghalliday — CHANGES_REQUESTED** [review]

**Comment by ghalliday** [comment]
@JamesDeFabia see comments

**Comment by JamesDeFabia** [comment]
@ghalliday Edits made, back to you

File: docs/ECLLanguageReference/ECLR_mods/Value-QString.xml [hunk 4]
```diff
@@ -23,6 +23,8 @@
   to the size needed to contain the result of a cast or passed parameter. You
   may use set indexing into any QSTRING to parse out a substring.</para>
 
+  <para>The upper size limit for any QSTRING value is 4GB.</para>
```
Comment (ghalliday) [inline docs/ECLLanguageReference/ECLR_mods/Value-QString.xml]: Technically the maximum length if 4G (=> maximum size is 3Gb), but I will merge anyway

**Review by ghalliday — APPROVED** [review]

**Pull request merged**